Dockerfile

```diff
@@ -55,6 +55,12 @@ COPY dashboard/src src
 COPY dashboard/package.json dashboard/yarn.lock ./
 RUN yarn install --production
 
+ARG IASQL_ENV=local
+ENV IASQL_ENV=$IASQL_ENV
+ENV NEXT_PUBLIC_IASQL_ENV=$IASQL_ENV
+
+RUN yarn build
```

Comment:
so then `IASQL_ENV` is now picked up when building the container and not running it?

Reply:
to preserve the behavior of https://github.com/iasql/iasql/pull/2039 the `IASQL_ENV` needs to move to the build command or it needs to be in both now?

Reply:
I am seeing the `to_upgrade` database back in posthog so there needs to be a change, hopefully it is only at build. In that case we should also throw if `IASQL_ENV` is passed at runtime so it doesn't get ignored or misused?

<img width="1141" alt="Screenshot 2023-02-22 at 7 21 43 AM" src="https://user-images.githubusercontent.com/2381303/220618521-f5e45498-8d53-4376-87e7-c24626696e5d.png">
ubusercontent.com/2381303/220618385-2d994bae-42b9-4c09-b89d-1794df393aab.png">

Reply:
Hmm yes, it would need to be sent on build time, the error might make sense, let me review that

Reply:
> I am seeing the `to_upgrade` database back in posthog so there needs to be a change, hopefully it is only at build. In that case we should also throw if `IASQL_ENV` is passed at runtime so it doesn't get ignored or misused?
> 
> <img alt="Screenshot 2023-02-22 at 7 21 43 AM" width="1141" src="https://user-images.githubusercontent.com/2381303/220618521-f5e45498-8d53-4376-87e7-c24626696e5d.png">

 This happened because I did not rebase. Now should not happen but you brought up a great point regarding the `IASQL_ENV`... I think we should just define a standard (maybe as you said just defining it at building time) to avoid misuse or confusion. 

Let me think a solution for this and will propose it before implementing it in another PR

Reply:
> - The NEXT_PUBLIC_IASQL_ENV env var still needs to be passed correctly on build time.

hmm if the above is true even if you rebase the upgrade tests will send metrics to posthog unless there is a change to the `docker-build-vol` command

Reply:
Ok, so I think the best solution is following the pattern I've added to this PR, request the `IASQL_ENV` to the API since it can know the env var value at run time and based on that pick the config object to use. It would not be exactly as is in different failed, probably will be a config object with the different envs and we will pick the one we need before rendering.

Reply:
it will always be taken into account

Reply:
I'm not sure what that means given there is build and runtime environment variable passing which opens several scenarios. Does this mean:
1) If `IASQL_ENV` is passed at runtime it will overwrite the one passed at build time?
2) If no `IASQL_ENV` is passed at runtime it will use the one at build time and not throw an error?
3) If no `IASQL_ENV` is passed at build time there will be an error thrown?

Reply:
There are facts to have the context:
- Docker defines [environment variables precedence](https://docs.docker.com/compose/environment-variables/envvars-precedence/) to override when needed. Env vars defined at runtime have more precedence than the ones defined at build time so they will be overwriten.

- We are not able to know when one env var has been defined at build or run time. We cannot know if we are overwriting a variable or not.

So answering your scenarios:

> 1. If IASQL_ENV is passed at runtime it will overwrite the one passed at build time?
Yes, always.

> 2. If no IASQL_ENV is passed at runtime it will use the one at build time and not throw an error?
Yes

> 3. If no IASQL_ENV is passed at build time there will be an error thrown?
Not sure if this is possible to do.


The solution proposed here is:
- Always define `IASQL_ENV=local` in the `Dockerfile`. This will have no implications in the build process of any of the projects and will provide the environment local if you run the container and do not pass any `IASQL_ENV` to `docker run` command.
- Define `IASQL_ENV` when running `docker run` to override the environment variable and use the desired value.